### ISSUE TITLE
onPlayerJoin and onQuit should return unregister callbacks

### ISSUE THREAD

**Issue by asadm** [asadm/playroom-unity#48]
The web versions return a function that can be called to unregister the listener

Related: https://discord.com/channels/997752993598419044/1224810795393613904/1224810795393613904

**Comment by momintlh** [comment]
After invoking the callback(s) we can maybe remove them from the maintained list similar to how RPC callbacks are being handled (or provide a function to remove / unregister)

**Comment by SaadBazaz** [comment]
> After invoking the callback(s) we can maybe remove them from the maintained list similar to how RPC callbacks are being handled

I don't think immediately removing them is the right idea.

> (or provide a function to remove / unregister)

This seems better. We can return a function which deletes the pointer/reference to that callback in the list (i.e. runs a `filter` on the list and removes the one stored in the function.

We would just have to cater for race conditions and make sure we handle errors/possible null operations correctly.

**Comment by SaadBazaz** [comment]
P.S.: A callback management mechanism was also discussed in #40.

**Comment by momintlh** [comment]
In #51 I have provided function to unsubscribe to listener (similar to the JS side), issue is we can't return a function back to unity to be called like this:
```c#
var unsubscribe = OnPlayerJoin(SomeCallcack())

unsubscribe()
```

current fix in #51 provides separate functions to unsubscribe to the listener (unity side). JS side is implemented similar to the docs.

**Comment by SaadBazaz** [comment]
Implemented in #51.
